src/librustc_typeck/diagnostics.rs

```diff
@@ -1611,9 +1611,9 @@ static BAR: _ = "test"; // error, explicitly write out the type instead
 "##,
 
 E0122: r##"
-An attempt was made to add a generic constraint to a type alias. While Rust will
-allow this with a warning, it will not currently enforce the constraint.
-Consider the example below:
+An attempt was made to add a generic constraint to a type alias. This constraint
+is entirely ignored. For backwards compatibility, Rust still allows this with a
+warning. Consider the example below:
```

Comment:
Yes, the constraint is only ignored when *referencing* the alias, not *within* it. Actually I think `<T as Trait>::Assoc` still requires a `T: Trait` bound. And the only reason that works is we effectively recover a `T: Trait` bound from the presence of a `<T as Trait>::Assoc` type, even if we don't know whether the type came from somewhere which *already required* `T: Trait`.